src/bridge/IrcHandler.ts

```diff
@@ -448,6 +448,55 @@ export class IrcHandler {
         }
     }
 
+    /**
+     * If configured, check to see if the all Matrix users in a given room are
+     * joined to a channel. If they are not, drop the message.
+     * @param req The IRC request
+     * @param server The IRC server.
+     */
+    private async requiresAllMatrixUsersJoined(server: IrcServer, channel: string, roomId: string): Promise<boolean> {
+        // The room state takes priority.
+        const stateRequires =
+            await this.ircBridge.roomConfigs.allowUnconnectedMatrixUsers(roomId, new IrcRoom(server, channel));
+        if (stateRequires !== null) {
+            return stateRequires;
+        }
+        return server.shouldRequireMatrixUserJoined(channel);
+    }
+
+    /**
+     * See if every joined Matrix user is also joined to the IRC channel. If they are not,
+     * this returns false. A seperate mechanism should be use to join the user if this fails.
+     * @param req The IRC request
+     * @param server The IRC server.
+     * @param channel The IRC channel.
+     * @param roomId The Matrix room.
+     * @returns True if all users are connected and joined, or false otherwise.
+     */
+    private async areAllMatrixUsersJoined(
+        req: BridgeRequest, server: IrcServer, channel: string, roomId: string): Promise<boolean> {
+        // Look for all the Matrix users in the room.
+        const members = await this.ircBridge.getMatrixUsersForRoom(roomId);
+        const pool = this.ircBridge.getClientPool();
+        for (const userId of members) {
+            if (userId === this.ircBridge.appServiceUserId) {
+                continue;
+            }
+            const client = pool.getBridgedClientByUserId(server, userId);
+            if (!client) {
+                req.log.warn(`${userId} has not connected to IRC yet, not bridging message`);
+                return false;
+            }
+            if (!client.inChannel(channel)) {
+                // TODO: Should we poke them into joining?
```

Comment:
Poke them? As in "sending them a message" that they need to join?

I guess the Matrix users in the room might be interested as to who is the problematic user that needs to be joined or kicked.

Reply:
More that we should just invoke a join, as this means the Matrix user is in the room but the bridge hasn't joined them.